src/util/ConfigManager/ConfigManager.h

```diff
@@ -186,10 +194,78 @@ class ConfigManager {
   */
   std::string printConfigurationDoc(bool printCurrentJsonConfiguration) const;
 
+  /*
+  @brief Add a function to the configuration manager for verifying, that the
+  values of the given configuration options are valid. Will always be called
+  after parsing.
+
+  @tparam ValidatorParameterTypes The types of the values in the
+  `validatorParameter`. Can be left up to type deduction.
+
+  @param validatorFunction Checks, if the values of the given configuration
+  options are valid. Should return true, if they are valid.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param validatorParameter Proxies for the configuration options, whos values
+  will be passed to the validator function as function arguments. Will keep the
+  same order.
+  */
+  template <typename... ValidatorParameterTypes>
+  void addValidator(
+      Validator<ValidatorParameterTypes...> auto validatorFunction,
+      std::string_view errorMessage,
+      ConfigOptionProxy<ValidatorParameterTypes>... validatorParameter)
```

Comment:
should be `configOptions` not `validatorParameter`. Or maybe better `configOptionsToBeChecked` (a long, but rather expressive name).

Reply:
Done, see [c34133d](https://github.com/ad-freiburg/qlever/pull/1058/commits/c34133d7923ca693cf58956f2ae76d69f5e6706d).